x/bank/client/testutil/suite.go

```diff
@@ -1,665 +0,0 @@
-package testutil
-
-import (
-	"fmt"
-	"io/ioutil"
-	"os"
-
-	"cosmossdk.io/math"
-	"github.com/gogo/protobuf/proto"
-	"github.com/stretchr/testify/suite"
-	tmcli "github.com/tendermint/tendermint/libs/cli"
-
-	"github.com/cosmos/cosmos-sdk/client/flags"
-	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
-	"github.com/cosmos/cosmos-sdk/testutil/network"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/cosmos/cosmos-sdk/types/query"
-	"github.com/cosmos/cosmos-sdk/x/bank/client/cli"
-	"github.com/cosmos/cosmos-sdk/x/bank/types"
-)
-
-type IntegrationTestSuite struct {
-	suite.Suite
-
-	cfg     network.Config
-	network *network.Network
-}
-
-func NewIntegrationTestSuite(cfg network.Config) *IntegrationTestSuite {
```

Comment:
one reason this test suite was not in a `_test.go` was to actually make it public for external chains (we're using it in Regen, Provenance wanted that, ref https://github.com/cosmos/cosmos-sdk/pull/6711#issuecomment-660291003).

Idea is that we can pass our own `NewRegenApp` into that test suite, and it will run all bank CLI tests, making sure we didn't mess up `NewRegenApp`.

I think that's still valuable, so I'm reluctant to move these tests into a `_test.go` file.

Reply:
So, the entire point of the EPIC is to make the CLI tests completely independent of simapp or any app for that matter and independent of Tendermint as well, so essentially they're just basic unit tests.

There is another EPIC that describes integration tests that should be more in the flavor of e2e tests. They would not exist in any module, but in `tests/e2e`.

Reply:
Should we not then move these to tests/e2e instead of deleting them? There is already the `cli_test.go` moved there. So when we refactor end-to-end tests, then we don't start from 0.

Reply:
OK I understand now what you're doing in this PR, it makes sense to decouple CLI tests from simapp 👍 

I guess I'm still confused what kind of tests `tests/e2e` will have: CLI? gRPC? (both of which are often closely tied to modules).

> There is another EPIC that describes integration tests that should be more in the flavor of e2e tests

Which EPIC is that?

Reply:
> I guess I'm still confused what kind of tests `tests/e2e` will have: CLI? gRPC? (both of which are often closely tied to modules).

We moved the current integration/e2e tests there and talked how this is a short term fix to unblock app wiring work. We still need to have a discussion around e2e tests. Bez proposed two ways, umee and osmosis. We should evaluate those and others to see what approach we should take.

Reply:
> Should we not then move these to tests/e2e instead of deleting them? There is already the cli_test.go moved there. So when we refactor end-to-end tests, then we don't start from 0.

No because I think they should be completely redefined or at least defined on an entire app-basis, not module by module. See https://github.com/cosmos/cosmos-sdk/issues/12527

> I guess I'm still confused what kind of tests tests/e2e will have: CLI? gRPC? (both of which are often closely tied to modules).

Most likely both -- gRPC and CLI

> Which EPIC is that?

I can't seem to find it. I think it was actually a comment in a PR by @kocubinski. @marbar3778 should we make one for e2e tests?

Reply:
I am also in favor of keeping (moving to `tests/e2e`) these tests until we have some consensus on levels of testing.  In our call this morning it was decided an ADR would be made to concentrate feedback on the topic.

Reply:
These are not how E2E tests should be designed IMO. I linked references to how I think we should design our E2E tests and they should be docker-based, not use an in-process Tendermint node.